lms/djangoapps/course_api/tests/test_views.py

```diff
@@ -142,5 +141,5 @@ def test_invalid_course_key(self):
         request = request_factory.get('/')
         request.query_params = {}
         request.user = self.staff_user
-        with self.assertRaises(NotFound):
-            CourseDetailView().get(request, 'a:b:c')
+        response = CourseDetailView().dispatch(request, course_key_string='a:b:c')
```

Comment:
Why call the view object directly instead of going through django's `reverse` and `client.get`?  We typically go through that path instead in test_views.py.